packages/server/src/Hocuspocus.ts

```diff
@@ -286,21 +298,30 @@ export class Hocuspocus {
   /**
    * Destroy the server
    */
-  async destroy(): Promise<any> {
-    this.server?.httpServer?.close()
+  destroy(): Promise<any> {
+    return new Promise(async resolve => {
 
-    try {
-      this.server?.webSocketServer?.close()
-      this.server?.webSocketServer?.clients.forEach(client => {
-        client.terminate()
-      })
-    } catch (error) {
-      console.error(error)
-    }
+      this.server?.httpServer?.close()
 
-    this.debugger.flush()
+      try {
+
+        this.configuration.extensions.push({
+          async afterUnloadDocument({ instance }) {
+            if (instance.getDocumentsCount() === 0) resolve('')
+          },
+        })
+
+        this.server?.webSocketServer?.close()
+        this.closeConnections()
+      } catch (error) {
+        console.error(error)
+      }
+
+      this.debugger.flush()
+
+      await this.hooks('onDestroy', { instance: this })
```

Comment:
This means the `destroy` method isn't waiting on the `onDestroy` hooks anymore - the `resolve` is called after all the documents are unloaded regardless of whether the `onDestroy` hooks have completed.

Reply:
true, thanks! Fixing in a new PR, as I can't update this one anymore: https://github.com/ueberdosis/hocuspocus/pull/750/files